mongo/data_store.go

```diff
@@ -195,26 +200,18 @@ func (m *JobStore) GetTasks(ctx context.Context, offsetParam string, limitParam
 		return results, nil
 	}
 
-	// Get all the tasks from the tasks collection and order them by lastupdated
-	iter := s.DB(m.Database).C(m.TasksCollection).Find(bson.M{"job_id": jobID}).Sort("last_updated").Iter()
-	defer func() {
-		err := iter.Close()
-		if err != nil {
-			log.Error(ctx, "error closing iterator", err)
-		}
-	}()
-
-	tasks := make([]models.Task, numTasks)
-	if err := iter.All(&tasks); err != nil {
-		return results, err
-	}
-
 	paginator := pagination.NewPaginator(m.cfg.DefaultLimit, m.cfg.DefaultOffset, m.cfg.DefaultMaxLimit)
```

Comment:
The pagination and validation shouldn't take place here, but should be called in the handlers for list endpoints.

Validation should always take place ahead of calling databases.

Reply:
That's ok but the reason why I put it here is because the ValidatePaginationParameters needs to use the number of tasks, which can only be found out by calling the database beforehand.

Reply:
I disagree here, I don't believe we need the number of tasks to validate the request. If the number of tasks is less than the offset and limit value then we would return status 200 with 0 items in response body.

Reply:
Ok, just to be clear then, if the user entered an offset of 5 but there were only 3 tasks then we should just return a 200 status and an empty list?

Reply:
I've made the changes Nathan - please have another look when you have time. Thanks.